lib/react.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const { plugin } = require('typescript-eslint')
+// @ts-ignore
+const reactPlugin = require('eslint-plugin-react')
+
+/**
+ * @typedef { 'recommended' | 'all' } ReactConfig
+ */
+
+/**
+ * @typedef ReactifyOptions
+ * @property {string[]|undefined} [ignores]
+ * @property {string} [name]
+ * @property {import('eslint').Linter.FlatConfig|undefined} tsConfig
+ * @property {ReactConfig|undefined} [reactConfig]
+ * @property {boolean} [requireReactImport]
+ */
+
+/**
+ * @param {ReactifyOptions} [options]
+ * @returns {Array<import('eslint').Linter.FlatConfig>}
+ */
+function reactify (options) {
+  const configName = options?.reactConfig || 'recommended'
+  const { ignores, name, tsConfig, requireReactImport } = options || {}
+
+  const reactConfig = reactPlugin.configs.flat[configName]
+  if (reactConfig == null || typeof reactConfig !== 'object') {
+    throw Error('Unexpected type of React ESLint config from "eslint-plugin-react"')
+  }
+
+  const jsxRuntimeConfig = !requireReactImport ? reactPlugin.configs.flat['jsx-runtime'] : {}
+  if (jsxRuntimeConfig == null || typeof jsxRuntimeConfig !== 'object') {
+    throw Error('Unexpected type of React ESLint config from "eslint-plugin-react"')
+  }
+
+  const settings = {
+    react: {
+      version: 'detect',
+    },
+  }
+
+  const reactConfigs = [
+    {
+      files: ['**/*.{jsx,cjsx,mjsx}'],
+      ...ignores && { ignores: ignores || [] },
+      name: name || `react/${configName}${!requireReactImport ? '+jsx-runtime' : ''}`,
+      ...{
+        ...reactConfig,
+        ...jsxRuntimeConfig,
+        rules: {
+          // ...reactConfig.rules,
+          ...jsxRuntimeConfig.rules,
+        },
+        settings,
+      },
+    },
+    ...tsConfig
+      ? [{
+          files: ['**/*.{tsx,ctsx,mtsx}'],
+          ...ignores && { ignores: ignores || [] },
+          name: name || `react-ts/${configName}${!requireReactImport ? '+jsx-runtime' : ''}`,
+          ...{
+            ...reactConfig,
+            rules: {
+              ...tsConfig.rules,
+              // ...reactConfig.rules,
+              ...jsxRuntimeConfig.rules,
+            },
+            plugins: {
+              ...reactConfig.plugins,
+              '@typescript-eslint': plugin,
+            },
+            languageOptions: {
+              ...tsConfig.languageOptions,
+              parserOptions: {
+                ...tsConfig.languageOptions?.parserOptions,
+                ...reactConfig.languageOptions.parserOptions,
+                ...jsxRuntimeConfig.languageOptions?.parserOptions,
+              },
+            },
+            settings,
+          },
+        }]
```

Comment:
Please do a deep merge of configs instead.

Reply:
Remove TSX support for now. TSX support will be part of another PR

Reply:
> Remove TSX support for now. TSX support should be part of another PR.

Doesn't necessarily have to be, but if it makes it simpler, then yeah

Reply:
I think a short discussion is needed on where to implement it:

- Do we extend JS config or `typescriptfied`  JS config? - the flat configuration priority ordering replication is somewhat lost then -
- Do we create a `reacttypescriptify()` helper if both react and ts are enabled? - the flat configuration ordering is kept - 


Either we settle on it for that PR, either we postpone it to another PR.